semgrep_output_v1.atd

```diff
@@ -606,14 +603,30 @@ type sca_match_kind = [
 ] < ocaml repr="classic">
 
 type transitive_reachable = {
-  (* TODO: some proof that the user can understand *)
-  explanation: string option;
+  (* The matches we found in 3rd party libraries.
+   * Ideally the location in cli_match are relative to the root of the project
+   * so one can display matches as package@/path/to/finding.py
+   * TODO: include matched code? 3rd party libraries are usually OSS
+   * so maybe ok to store code for once in our DBs? Otherwise The App will
+   * have to redownload and reaccess the package code (or maybe we
+   * could give a github URL?)
+   *)
+  matches: (string (* package uri? *) * cli_match list) list;
+  (* LATER: add callgraph information so one can see the path from 1st party code
+   * to the vulnerable intermediate 3rd party function.
+   * This is set to None for now.
+   *)
   callgraph_reachable: bool option;
-  (* LATER: add trace information *)
+  (* some extra explanation that the user can understand *)
+  explanation: string option;
 }
 
 type transitive_unreachable = {
-  (* TODO: some proof that the user can understand *)
+  (* We didn't find any findings in all the 3rd party libraries that are using
+   * the 3rd party vulnerable library. This is a "proof of work".
+   *)
+  analyzed_packages: string (* package uri? *) list;
```

Comment:
I'm iffy on using just a string to identify packages. It's going to be hard to find the package name and version from that, which is what we probably want to display to the user. Maybe we should use `found_dependency` here? Or maybe just name/version/ecosystem triplet?